src/types/decimal.rs

```diff
@@ -514,6 +593,100 @@ mod decimal_tests {
         assert_eq!(actual, expected);
     }
 
+    #[test]
+    fn test_difference_by_parts() {
+        let d1: Decimal = Decimal::new(123456789, -2);
+        let d10: Decimal = Decimal::new(123456789, -1);
+        let d100: Decimal = Decimal::new(123456789, 0);
+
+        // diff d1 with d1
+        let (diff_integer, diff_fractional) = Decimal::difference_by_parts(d1.clone(), d1.clone());
```

Comment:
Can we make `difference_by_parts` take `&Decimal`s and `clone()` them itself? The cloning seems like something that we might want to change/optimize later.